### ISSUE TITLE
Feature: Bring permutation API closer to hf/torch style dataset manipulation apis

### ISSUE THREAD

**Issue by AyushExel** [lancedb/lancedb#3244]
### SDK

Python

###  2-step permutation might not intuitive for existing torch/hf users. 

PermutationBuilder and Permutation are two separate classes with overlapping concerns. The builder handles row selection (shuffle, filter, split); Permutation handles read-time concerns (column selection, format, transform). This distinction matters to a data engineer but is invisible and irrelevant to an ML engineer loading a training set. 
```
# Today: Permutation.shuffle() raises AttributeError
perm_tbl = permutation_builder(table).shuffle().execute()
perm = Permutation.from_tables(table, perm_tbl)

# What ML engineers might expect
perm = Permutation.from_table(table).shuffle()
```
